xapian-letor/api/featurelist.cc

```diff
@@ -100,11 +103,14 @@ FeatureList::create_feature_vectors(const Xapian::MSet & mset, const Xapian::Que
     for (Xapian::MSetIterator i = mset.begin(); i != mset.end(); ++i) {
 	Xapian::Document doc = i.get_document();
 	std::vector<double> fVals;
-
+	// All stats are computed and stored in the corresponding variables.
+	internal->compute_statistics(letor_query, letor_db, doc);
 	for (std::vector<Feature*>::const_iterator it = feature.begin(); it != feature.end(); ++it) {
 	    (*it)->set_database(letor_db);
 	    (*it)->set_query(letor_query);
 	    (*it)->set_doc(doc);
+	    // Populates the Feature with required stats
+	    internal->populate_feature(*it);
 	    vector<double> values = (*it)->get_values();
```

Comment:
It's better to use a const reference here as it helps the compiler avoid a copy (and copying a vector is not cheap):

```
const vector<double>& values = (*it)->get_values();
```

Assigning a temporary to a const reference like this is special in C++ and means it lives for the scope of the reference - see https://herbsutter.com/2008/01/01/gotw-88-a-candidate-for-the-most-important-const/ for example.  Note especially that a non-const reference doesn't work.